### ISSUE TITLE
Effect of second compile for model

### ISSUE THREAD

**Issue by dr-costas** [fchollet/keras#6164]
Hi, 

Suppose that someone builds a model. Then he will compile it with a loss L_1 and a optimizer Opt_1. If, at a later point and after some training, he re-compiles the model with optimizer Opt_2, what will be the effect on the model? 

Will the weights re-initialized? Or the initialization in the build method will be kept and, thus, the weights will be the ones calculated by the Opt_1 optimizer?

**Comment by dr-costas** [comment]
Just tried it. 

The weights will remain the same. 

Sample code: 

```python
import numpy as np
from keras.layers import Input, GRU

b_size = 32
t_steps = 10
feats_in = 5
feats_out = 3

x = np.random.randn(b_size, t_steps, feats_in)
y = np.random.randn(b_size, t_steps, feats_out)

i = Input(shape=(t_steps, feats_in))
l = GRU(feats_out, return_sequences=True)(i)

m = Model(i, l)
m.compile(loss='mse', optimizer='sgd')
m.fit(x, y)

w = m.get_weights()
m.compile(loss='mse', optimizer='adam')
w2 = m.get_weights()

print(all([np.allclose(i1, i2) for i1, i2 in zip(w, w2)]))
```

**Comment by fchollet** [comment]
Just, `compile` does not change the state of the model.
